c2_components/src/mfx_c2_decoder_component.cpp

```diff
@@ -1492,12 +1523,50 @@ c2_status_t MfxC2DecoderComponent::AllocateFrame(MfxC2FrameOut* frame_out)
                     /* Buffer locked, try next block. */
                     MFX_DEBUG_TRACE_PRINTF("Buffer still locked, try next block");
                     res = C2_TIMED_OUT;
-                }
-                else
+                } else {
                     *frame_out = MfxC2FrameOut(std::move(out_block), it->second);
+                }
             }
         } else {
-            res = MfxC2FrameOut::Create(converter, out_block, m_mfxVideoParams.mfx.FrameInfo, TIMEOUT_NS, frame_out, hndl.get());
+            if (m_mfxVideoParams.mfx.FrameInfo.Width >= WIDTH_2K || m_mfxVideoParams.mfx.FrameInfo.Height >= HEIGHT_2K) {
+                // Thumbnail generation for 4K/8K video
```

Comment:
for 4K/8K? but this jugde include 2K?

Reply:
2K thumbnail generation can also this logic. But generally, sw codec2.0 can support up to 2K decoding. If sw codec2.0 failed to decode the thumbnail, msdk codec2.0 can be used.